trpcultivate_phenotypes/tests/src/Kernel/Display/DisplayValidationResultWindowTest.php

```diff
@@ -0,0 +1,406 @@
+<?php
+
+namespace Drupal\Tests\trpcultivate_phenotypes\Kernel\Display\DisplayValidationResultWindowTest;
+
+use Drupal\Tests\tripal_chado\Kernel\ChadoTestKernelBase;
+
+/**
+ * Tests Tripal Cultivate Phenotypes Validation Result Window.
+ *
+ * @group trpcultivate_phenotypes
+ * @group displays
+ */
+class DisplayValidationResultWindowTest extends ChadoTestKernelBase {
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  protected static $modules = [
+    'tripal',
+    'tripal_chado',
+    'trpcultivate_phenotypes',
+  ];
+
+  /**
+   * Drupal render service.
+   *
+   * @var Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Set test environment.
+    \Drupal::state()->set('is_a_test_environment', TRUE);
+
+    $this->renderer = \Drupal::service('renderer');
+  }
+
+  /**
+   * Data Provider: Provides the template with a set of validation item values.
+   *
+   * @return array
+   *   Each scenario/element is an array with the following values.
+   *   - A string, human-readable short description of the test scenario.
+   *   - An array that contains a set of values with each set containing the
+   *     keys title, status and details as returned by a validator. The details
+   *     key contains values of the failed items structured as
+   *     a Drupal renderarray.
+   *   - The expected class name an item is tagged with. Each validation item
+   *     corresponds to one class name based on the the value of the status key.
```

Comment:
I borrowed your description of the validation status array from your twig file for consistency :)

```suggestion
   *   - One or more validation status arrays. Each array contains the following
   *     keys:
   *     - 'title': Validation title text that is paired with appropriate status
   *        icon in the markup output.
   *     - 'status': The status of the validation that indicates if it passed 
   *        (pass), failed (fail) or upcoming task (todo).
   *     - 'details': A Drupal render array that will generate the required markup
   *        to output any failed items, only if 'status' is set to 'fail'.
   *   - The expected class name an item is tagged with. Each validation item
   *     corresponds to one class name based on the value of the status key.
```